File: src/static_tf/src/publisher.cpp
// Publisher of static coordinate transformation:
//     Publishes position information about the laser coordinate system 
// Implementation steps:
//     1.Import module
//     2.Initialize ROS node
//     3.Create static coordinate broadcaster
//     4.Create and organize the broadcast message
//     5.The broadcaster sends a message
//     6.spinOnce()

#include "ros/ros.h"
#include "tf2_ros/static_transform_broadcaster.h"
#include "geometry_msgs/TransformStamped.h"
#include "tf2/LinearMath/Quaternion.h"

int main(int argc, char *argv[])
{

    ros::init(argc, argv, "static_broadcast");

    tf2_ros::StaticTransformBroadcaster broadcaster;

    geometry_msgs::TransformStamped ts;

    ts.header.seq = 100;
    ts.header.stamp = ros::Time::now();
    ts.header.frame_id = "base_link";

    ts.child_frame_id = "laser";

    ts.transform.translation.x = 0.2;
    ts.transform.translation.y = 0.0;
    ts.transform.translation.z = 0.5;

    tf2::Quaternion qtn;
    qtn.setRPY(0,0,0);
    ts.transform.rotation.x = qtn.getX();
    ts.transform.rotation.y = qtn.getY();
    ts.transform.rotation.z = qtn.getZ();
    ts.transform.rotation.w = qtn.getW();

    broadcaster.sendTransform(ts);
    ros::spin();
    return 0;
}

File: src/static_tf/scripts/publisher.py
#! /usr/bin/env python3

"""  
    Publisher of static coordinate transformation:
        Publishes position information about the laser coordinate system 
    Implementation steps:
        1.Import module
        2.Initialize ROS node
        3.Create static coordinate broadcaster
        4.Create and organize the broadcast message
        5.The broadcaster sends a message
        6.spinOnce()
"""

import rospy
import tf2_ros
import tf
from geometry_msgs.msg import TransformStamped

if __name__ == "__main__":

    rospy.init_node("static_tf_pub_p")

    broadcaster = tf2_ros.StaticTransformBroadcaster()

    tfs = TransformStamped()
    # header
    tfs.header.frame_id = "world"
    tfs.header.stamp = rospy.Time.now()
    tfs.header.seq = 101
    # child coordinate
    tfs.child_frame_id = "radar"
    # relative information of coordinate system
    tfs.transform.translation.x = 0.2
    tfs.transform.translation.y = 0.0
    tfs.transform.translation.z = 0.5

    qtn = tf.transformations.quaternion_from_euler(0,0,0)
    tfs.transform.rotation.x = qtn[0]
    tfs.transform.rotation.y = qtn[1]
    tfs.transform.rotation.z = qtn[2]
    tfs.transform.rotation.w = qtn[3]

    broadcaster.sendTransform(tfs)

    rospy.spin()